templates/kubernetes-cluster.template

```diff
@@ -422,6 +409,59 @@ Resources:
           - LoadBalancerHostname: !GetAtt ApiLoadBalancer.DNSName
             LoadBalancerName: !Ref ApiLoadBalancer
 
+  # IAM role for Lambda function for generating kubeadm token
+  LambdaExecutionRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+        - Effect: "Allow"
+          Principal:
+            Service: ["lambda.amazonaws.com"]
+          Action: "sts:AssumeRole"
+      Path: "/"
+      Policies:
+      - PolicyName: "lambda_policy"
+        PolicyDocument:
+          Version: "2012-10-17"
+          Statement:
+          - Effect: "Allow"
+            Action:
+            - "logs:CreateLogGroup"
+            - "logs:CreateLogStream"
+            - "logs:PutLogEvents"
+            Resource: "arn:aws:logs:*:*:*"
+
+  # Lambda Function for generating the kubeadm token
+  GenKubeadmToken:
+    Type: "AWS::Lambda::Function"
+    Properties:
+      Code:
+        ZipFile: |
+          import random
+          import cfnresponse
+          def handler(event, context):
+            if event['RequestType'] == 'Delete':
+              cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
+            if event['RequestType'] == 'Create':
+              token = ("%06x.%016x" % (random.SystemRandom().getrandbits(3*8), random.SystemRandom().getrandbits(8*8)))
```

Comment:
This produces a hex string, which is lower entropy than mixed upper/lower/numeric characters, which is what kubeadm supports.

I'm not a python guy at all but just trolling SO for an easy answer: (http://stackoverflow.com/questions/2257441/random-string-generation-with-upper-case-letters-and-digits-in-python)

```
require string
require random
...
def id_generator(size=6, chars=string.ascii_uppercase + string.ascii_lowercase + string.digits):
  return ''.join(random.choice(chars) for _ in range(size))
...
token = ("%s.%s" % (id_generator(6), id_generator(16)))
```

that appears to work with full alphanumeric strings.